Major lag/Throttling when chunks finishes generating and runs out of memory when leaving world

Currently running on Linux with the Prism Launcher flatpak with both voxy rendering at 96 chunks and worldgen with generation radius at 96.
FPS with the mod disabled through the mod menu:
<img width="3840" height="2160" alt="Image" src="https://github.com/user-attachments/assets/e6d649fc-4d80-4295-8322-ae14dee20502" />
FPS with the mod enabled, after chunks finishes generating:
<img width="3840" height="2160" alt="Image" src="https://github.com/user-attachments/assets/979abcfc-57a3-480d-9702-b52a4fd9bbaf" />
The mod reports that it is being throttled for some reason, nvtop reports low GPU usage while VRAM usage is very high.
When leaving the world, the game runs out of memory:
<img width="3840" height="2160" alt="Image" src="https://github.com/user-attachments/assets/006e8b6d-aefc-41ad-ad2c-bf862c1b79e4" />
I've already tried running with 24 GB allocated, but it still consumed all the RAM I allocated.
```
Prism Launcher version: 10.0.5 (flatpak)

Java is version 21.0.7, using 64 (amd64) architecture, from Microsoft.


AMD Ryzen 7 5700G with Radeon Graphics
NVIDIA Corporation GB206 [GeForce RTX 5060] (rev a1)

Subsystem: Micro-Star International Co., Ltd. [MSI] Device 5371

Kernel driver in use: nvidia

Advanced Micro Devices, Inc. [AMD/ATI] Cezanne [Radeon Vega Series / Radeon Vega Mobile Series] (rev c8)

Subsystem: Gigabyte Technology Co., Ltd Device d000

Kernel driver in use: amdgpu

OpenGL version string: 4.6 (Compatibility Profile) Mesa 25.3.3 (git-a9c6420d8c)

Main Class:
  net.fabricmc.loader.impl.launch.knot.KnotClient

Native path:
  /home/********/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/instances/MixedPack/natives

Traits:
traits feature:is_quick_play_multiplayer
traits feature:is_quick_play_singleplayer
traits FirstThreadOnMacOS
traits XR:Initial

Libraries:
  /home/********/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/libraries/org/lwjgl/lwjgl-freetype-natives-linux/3.3.3/lwjgl-freetype-natives-linux-3.3.3.jar
  /home/********/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/libraries/org/lwjgl/lwjgl-freetype/3.3.3/lwjgl-freetype-3.3.3.jar
  /home/********/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/libraries/org/lwjgl/lwjgl-glfw-natives-linux/3.3.3/lwjgl-glfw-natives-linux-3.3.3.jar
  /home/********/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/libraries/org/lwjgl/lwjgl-glfw/3.3.3/lwjgl-glfw-3.3.3.jar
  /home/********/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/libraries/org/lwjgl/lwjgl-jemalloc-natives-linux/3.3.3/lwjgl-jemalloc-natives-linux-3.3.3.jar
  /home/********/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/libraries/org/lwjgl/lwjgl-jemalloc/3.3.3/lwjgl-jemalloc-3.3.3.jar
  /home/********/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/libraries/org/lwjgl/lwjgl-natives-linux/3.3.3/lwjgl-natives-linux-3.3.3.jar
  /home/********/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/libraries/org/lwjgl/lwjgl-openal-natives-linux/3.3.3/lwjgl-openal-natives-linux-3.3.3.jar
  /home/********/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/libraries/org/lwjgl/lwjgl-openal/3.3.3/lwjgl-openal-3.3.3.jar
  /home/********/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/libraries/org/lwjgl/lwjgl-opengl-natives-linux/3.3.3/lwjgl-opengl-natives-linux-3.3.3.jar
  /home/********/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/libraries/org/lwjgl/lwjgl-opengl/3.3.3/lwjgl-opengl-3.3.3.jar
  /home/********/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/libraries/org/lwjgl/lwjgl-stb-natives-linux/3.3.3/lwjgl-stb-natives-linux-3.3.3.jar
  /home/********/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/libraries/org/lwjgl/lwjgl-stb/3.3.3/lwjgl-stb-3.3.3.jar
  /home/********/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/libraries/org/lwjgl/lwjgl-tinyfd-natives-linux/3.3.3/lwjgl-tinyfd-natives-linux-3.3.3.jar
  /home/********/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/libraries/org/lwjgl/lwjgl-tinyfd/3.3.3/lwjgl-tinyfd-3.3.3.jar
  /home/********/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/libraries/org/lwjgl/lwjgl/3.3.3/lwjgl-3.3.3.jar
  /home/********/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/libraries/at/yawk/lz4/lz4-java/1.8.1/lz4-java-1.8.1.jar
  /home/********/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/libraries/com/azure/azure-json/1.4.0/azure-json-1.4.0.jar
  /home/********/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/libraries/com/github/oshi/oshi-core/6.9.0/oshi-core-6.9.0.jar
  /home/********/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/libraries/com/google/code/gson/gson/2.13.2/gson-2.13.2.jar
  /home/********/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/libraries/com/google/guava/failureaccess/1.0.3/failureaccess-1.0.3.jar
  /home/********/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/libraries/com/google/guava/guava/33.5.0-jre/guava-33.5.0-jre.jar
  /home/********/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/libraries/com/ibm/icu/icu4j/77.1/icu4j-77.1.jar
  /home/********/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/libraries/com/microsoft/azure/msal4j/1.23.1/msal4j-1.23.1.jar
  /home/********/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/libraries/com/mojang/authlib/7.0.61/authlib-7.0.61.jar
  /home/********/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/libraries/com/mojang/blocklist/1.0.10/blocklist-1.0.10.jar
  /home/********/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/libraries/com/mojang/brigadier/1.3.10/brigadier-1.3.10.jar
  /home/********/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/libraries/com/mojang/datafixerupper/9.0.19/datafixerupper-9.0.19.jar
  /home/********/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/libraries/com/mojang/jtracy/1.0.37/jtracy-1.0.37.jar
  /home/********/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/libraries/com/mojang/jtracy-natives-linux/1.0.37/jtracy-natives-linux-1.0.37.jar
  /home/********/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/libraries/com/mojang/logging/1.6.11/logging-1.6.11.jar
  /home/********/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/libraries/com/mojang/patchy/2.2.10/patchy-2.2.10.jar
  /home/********/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/libraries/com/mojang/text2speech/1.18.11/text2speech-1.18.11.jar
  /home/********/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/libraries/commons-codec/commons-codec/1.19.0/commons-codec-1.19.0.jar
  /home/********/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/libraries/commons-io/commons-io/2.20.0/commons-io-2.20.0.jar
  /home/********/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/libraries/io/netty/netty-buffer/4.2.7.Final/netty-buffer-4.2.7.Final.jar
  /home/********/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/libraries/io/netty/netty-codec-base/4.2.7.Final/netty-codec-base-4.2.7.Final.jar
  /home/********/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/libraries/io/netty/netty-codec-compression/4.2.7.Final/netty-codec-compression-4.2.7.Final.jar
  /home/********/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/libraries/io/netty/netty-codec-http/4.2.7.Final/netty-codec-http-4.2.7.Final.jar
  /home/********/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/libraries/io/netty/netty-common/4.2.7.Final/netty-common-4.2.7.Final.jar
  /home/********/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/libraries/io/netty/netty-handler/4.2.7.Final/netty-handler-4.2.7.Final.jar
  /home/********/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/libraries/io/netty/netty-resolver/4.2.7.Final/netty-resolver-4.2.7.Final.jar
  /home/********/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/libraries/io/netty/netty-transport-classes-epoll/4.2.7.Final/netty-transport-classes-epoll-4.2.7.Final.jar
  /home/********/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/libraries/io/netty/netty-transport-classes-kqueue/4.2.7.Final/netty-transport-classes-kqueue-4.2.7.Final.jar
  /home/********/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/libraries/io/netty/netty-transport-native-epoll/4.2.7.Final/netty-transport-native-epoll-4.2.7.Final-linux-aarch_64.jar
  /home/********/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/libraries/io/netty/netty-transport-native-epoll/4.2.7.Final/netty-transport-native-epoll-4.2.7.Final-linux-x86_64.jar
  /home/********/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/libraries/io/netty/netty-transport-native-unix-common/4.2.7.Final/netty-transport-native-unix-common-4.2.7.Final.jar
  /home/********/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/libraries/io/netty/netty-transport/4.2.7.Final/netty-transport-4.2.7.Final.jar
  /home/********/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/libraries/it/unimi/dsi/fastutil/8.5.18/fastutil-8.5.18.jar
  /home/********/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/libraries/net/java/dev/jna/jna-platform/5.17.0/jna-platform-5.17.0.jar
  /home/********/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/libraries/net/java/dev/jna/jna/5.17.0/jna-5.17.0.jar
  /home/********/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/libraries/net/sf/jopt-simple/jopt-simple/5.0.4/jopt-simple-5.0.4.jar
  /home/********/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/libraries/org/apache/commons/commons-compress/1.28.0/commons-compress-1.28.0.jar
  /home/********/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/libraries/org/apache/commons/commons-lang3/3.19.0/commons-lang3-3.19.0.jar
  /home/********/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/libraries/org/apache/logging/log4j/log4j-api/2.25.2/log4j-api-2.25.2.jar
  /home/********/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/libraries/org/apache/logging/log4j/log4j-core/2.25.2/log4j-core-2.25.2.jar
  /home/********/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/libraries/org/apache/logging/log4j/log4j-slf4j2-impl/2.25.2/log4j-slf4j2-impl-2.25.2.jar
  /home/********/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/libraries/org/jcraft/jorbis/0.0.17/jorbis-0.0.17.jar
  /home/********/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/libraries/org/joml/joml/1.10.8/joml-1.10.8.jar
  /home/********/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/libraries/org/jspecify/jspecify/1.0.0/jspecify-1.0.0.jar
  /home/********/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/libraries/org/slf4j/slf4j-api/2.0.17/slf4j-api-2.0.17.jar
  /home/********/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/libraries/net/fabricmc/intermediary/1.21.11/intermediary-1.21.11.jar
  /home/********/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/libraries/org/ow2/asm/asm/9.9/asm-9.9.jar
  /home/********/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/libraries/org/ow2/asm/asm-analysis/9.9/asm-analysis-9.9.jar
  /home/********/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/libraries/org/ow2/asm/asm-commons/9.9/asm-commons-9.9.jar
  /home/********/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/libraries/org/ow2/asm/asm-tree/9.9/asm-tree-9.9.jar
  /home/********/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/libraries/org/ow2/asm/asm-util/9.9/asm-util-9.9.jar
  /home/********/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/libraries/net/fabricmc/sponge-mixin/0.17.0+mixin.0.8.7/sponge-mixin-0.17.0+mixin.0.8.7.jar
  /home/********/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/libraries/net/fabricmc/fabric-loader/0.18.4/fabric-loader-0.18.4.jar
  /home/********/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/libraries/com/mojang/minecraft/1.21.11/minecraft-1.21.11-client.jar

Native libraries:

Mods:
  [✔] CombatNouveau-v21.11.1-mc1.21.11-Fabric
  [✔] ForgeConfigAPIPort-v21.11.1-mc1.21.11-Fabric
  [✔] LeavesBeGone-v21.11.0-mc1.21.11-Fabric
  [✔] PuzzlesLib-v21.11.11-mc1.21.11-Fabric
  [✔] TaxFreeLevels-1.4.16-fabric-1.21.11
  [✔] Voxy World Gen V2-1.21.11-2.2.2
  [✔] bedrockify-1.11.5+mc1.21.11
  [✔] cloth-config-21.11.153-fabric
  [✔] continuity-3.0.1-beta.1+1.21.11
  [✔] cwb-fabric-3.0.0-build.14+mc1.21.11
  [✔] fabric-api-0.141.3+1.21.11
  [✔] fabric-language-kotlin-1.13.9+kotlin.2.3.10
  [✔] ferritecore-8.2.0-fabric
  [✔] hardcover-9.0.0+1.21.11
  [✔] hunger-remover-1.1.11
  [✔] iris-fabric-1.10.5+mc1.21.11
  [✔] midnightlib-fabric-1.9.2+1.21.11
  [✔] moderner-beta-fabric-4.1.0+1.21.11
  [✔] modmenu-17.0.0-beta.2
  [✔] moreculling-fabric-1.21.11-1.6.1
  [✔] owo-lib-0.13.0+1.21.11
  [✔] placeholder-api-2.8.2+1.21.10
  [✔] skinlayers3d-fabric-1.10.1-mc1.21.11
  [✔] smallviewmodel-2.0.1
  [✔] sodium-fabric-0.8.4+mc1.21.11
  [✔] voxy-0.2.10-alpha

Params:
  --username  --version 1.21.11 --gameDir /home/********/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/instances/MixedPack/minecraft --assetsDir /home/********/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/assets --assetIndex 29 --uuid  --accessToken  --versionType release

Window size: 854 x 480

Launcher: standard

Java Arguments:
[-Xms512m, -Xmx12288m, -Duser.language=en]


Minecraft process ID: 27901


[17:49:37] [main/INFO]: Loading Minecraft 1.21.11 with Fabric Loader 0.18.4
[17:49:37] [main/INFO]: Loading 114 mods:
	- bedrockify 1.11.5+mc1.21.11
	- cloth-config 21.11.153
	   \-- cloth-basic-math 0.6.1
	- combatnouveau 21.11.1
	- continuity 3.0.1-beta.1+1.21.11
	- cubes-without-borders 3.0.0-build.14+mc1.21.11
	- fabric-api 0.141.3+1.21.11
	   |-- fabric-api-base 1.0.5+4ebb5c083e
	   |-- fabric-api-lookup-api-v1 1.6.114+20dc27073e
	   |-- fabric-biome-api-v1 17.1.1+4fc5413f3e
	   |-- fabric-block-api-v1 1.1.10+4ebb5c083e
	   |-- fabric-block-view-api-v2 1.0.39+4ebb5c083e
	   |-- fabric-command-api-v2 2.4.7+6b42a6003e
	   |-- fabric-content-registries-v0 10.2.14+4fc5413f3e
	   |-- fabric-convention-tags-v1 2.1.55+7f945d5b3e
	   |-- fabric-convention-tags-v2 2.17.3+8ef948ba3e
	   |-- fabric-crash-report-info-v1 0.3.23+4ebb5c083e
	   |-- fabric-data-attachment-api-v1 1.8.47+eed0806f3e
	   |-- fabric-data-generation-api-v1 23.4.0+69974c4e3e
	   |-- fabric-dimensions-v1 4.0.28+4fc5413f3e
	   |-- fabric-entity-events-v1 3.1.1+1d0ab4303e
	   |-- fabric-events-interaction-v0 4.1.0+3b89ecf63e
	   |-- fabric-game-rule-api-v1 2.0.3+4fc5413f3e
	   |-- fabric-item-api-v1 11.5.20+d0c46b9e3e
	   |-- fabric-item-group-api-v1 4.2.36+4fc5413f3e
	   |-- fabric-key-binding-api-v1 1.1.7+4fc5413f3e
	   |-- fabric-lifecycle-events-v1 2.6.15+4ebb5c083e
	   |-- fabric-loot-api-v2 3.0.73+3f89f5a53e
	   |-- fabric-loot-api-v3 2.0.20+78c8b4663e
	   |-- fabric-message-api-v1 6.1.12+4ebb5c083e
	   |-- fabric-model-loading-api-v1 6.0.15+4fc5413f3e
	   |-- fabric-networking-api-v1 5.1.5+ae1e07683e
	   |-- fabric-object-builder-api-v1 21.1.39+4fc5413f3e
	   |-- fabric-particles-v1 4.2.11+4fc5413f3e
	   |-- fabric-recipe-api-v1 8.2.3+4ebb5c083e
	   |-- fabric-registry-sync-v0 6.2.5+1718722b3e
	   |-- fabric-renderer-api-v1 8.0.3+f4ffd2e53e
	   |-- fabric-renderer-indigo 5.0.3+f4ffd2e53e
	   |-- fabric-rendering-fluids-v1 3.1.43+4ebb5c083e
	   |-- fabric-rendering-v1 16.2.10+0290ad933e
	   |-- fabric-resource-conditions-api-v1 5.0.35+4fc5413f3e
	   |-- fabric-resource-loader-v0 3.3.4+4fc5413f3e
	   |-- fabric-resource-loader-v1 1.0.10+78c8b4663e
	   |-- fabric-screen-api-v1 3.1.7+4ebb5c083e
	   |-- fabric-screen-handler-api-v1 1.3.161+4fc5413f3e
	   |-- fabric-serialization-api-v1 1.0.5+4ebb5c083e
	   |-- fabric-sound-api-v1 1.0.51+4fc5413f3e
	   |-- fabric-tag-api-v1 1.2.20+4fc5413f3e
	   |-- fabric-transfer-api-v1 6.0.25+4fc5413f3e
	   \-- fabric-transitive-access-wideners-v1 7.1.0+014c8cec3e
	- fabric-language-kotlin 1.13.9+kotlin.2.3.10
	   |-- org_jetbrains_kotlin_kotlin-reflect 2.3.10
	   |-- org_jetbrains_kotlin_kotlin-stdlib 2.3.10
	   |-- org_jetbrains_kotlin_kotlin-stdlib-jdk7 2.3.10
	   |-- org_jetbrains_kotlin_kotlin-stdlib-jdk8 2.3.10
	   |-- org_jetbrains_kotlinx_atomicfu-jvm 0.31.0
	   |-- org_jetbrains_kotlinx_kotlinx-coroutines-core-jvm 1.10.2
	   |-- org_jetbrains_kotlinx_kotlinx-coroutines-jdk8 1.10.2
	   |-- org_jetbrains_kotlinx_kotlinx-datetime-jvm 0.7.1
	   |-- org_jetbrains_kotlinx_kotlinx-io-bytestring-jvm 0.8.2
	   |-- org_jetbrains_kotlinx_kotlinx-io-core-jvm 0.8.2
	   |-- org_jetbrains_kotlinx_kotlinx-serialization-cbor-jvm 1.10.0
	   |-- org_jetbrains_kotlinx_kotlinx-serialization-core-jvm 1.10.0
	   \-- org_jetbrains_kotlinx_kotlinx-serialization-json-jvm 1.10.0
	- fabricloader 0.18.4
	   \-- mixinextras 0.5.0
	- ferritecore 8.2.0
	- forgeconfigapiport 21.11.1
	   |-- com_electronwill_night-config_core 3.8.3
	   \-- com_electronwill_night-config_toml 3.8.3
	- hardcover 9.0.0+1.21.11
	- hunger_remover 1.1.11
	- iris 1.10.5+mc1.21.11
	   |-- io_github_douira_glsl-transformer 3.0.0-pre3
	   |-- org_anarres_jcpp 1.4.14
	   \-- org_antlr_antlr4-runtime 4.13.1
	- java 21
	- leavesbegone 21.11.0
	- midnightlib 1.9.2
	- minecraft 1.21.11
	- moderner_beta 4.1.0+1.21.11
	- modmenu 17.0.0-beta.2
	- moreculling 1.6.1
	   |-- conditional-mixin 0.6.3
	   \-- mixinsquared 0.2.0
	- owo 0.13.0+1.21.11
	   |-- blue_endless_jankson 1.2.2
	   |-- com_github_kdl-org_kdl4j 1.0.1
	   |-- io_wispforest_endec 0.1.12
	   |-- io_wispforest_endec_gson 0.1.7
	   |-- io_wispforest_endec_jankson 0.1.7
	   \-- io_wispforest_endec_netty 0.1.6
	- placeholder-api 2.8.2+1.21.10
	- puzzleslib 21.11.11
	- skinlayers3d 1.10.1
	   |-- transition 1.0.11
	   \-- trender 1.0.10
	- smallviewmodel 2.0.1
	   \-- lattice 1.3.1
	- sodium 0.8.4+mc1.21.11
	- taxfreelevels 1.4.16
	- voxy 0.2.10-alpha
	   |-- org_apache_commons_commons-pool2 2.12.0
	   |-- org_lwjgl_lwjgl-lmdb 3.3.3
	   |-- org_lwjgl_lwjgl-lmdb_natives-linux 3.3.3
	   |-- org_lwjgl_lwjgl-lmdb_natives-windows 3.3.3
	   |-- org_lwjgl_lwjgl-zstd 3.3.3
	   |-- org_lwjgl_lwjgl-zstd_natives-linux 3.3.3
	   |-- org_lwjgl_lwjgl-zstd_natives-windows 3.3.3
	   |-- org_lz4_lz4-java 1.8.0
	   |-- org_rocksdb_rocksdbjni 10.2.1
	   |-- org_tukaani_xz 1.10
	   \-- redis_clients_jedis 5.1.0
	- voxyworldgenv2 2.2.2
[17:49:38] [main/INFO]: SpongePowered MIXIN Subsystem Version=0.8.7 Source=file:/home/********/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/libraries/net/fabricmc/sponge-mixin/0.17.0+mixin.0.8.7/sponge-mixin-0.17.0+mixin.0.8.7.jar Service=Knot/Fabric Env=CLIENT
[17:49:38] [main/INFO]: Compatibility level set to JAVA_16
[17:49:38] [main/INFO]: Compatibility level set to JAVA_21
[17:49:38] [main/WARN]: Reference map 'iris.refmap.json' for mixins.iris.json could not be read. If this is a development environment you can ignore this message
[17:49:38] [main/WARN]: Reference map 'iris-fabric.refmap.json' for mixins.iris.fabric.json could not be read. If this is a development environment you can ignore this message
[17:49:38] [main/WARN]: Reference map 'iris.refmap.json' for mixins.iris.fantastic.json could not be read. If this is a development environment you can ignore this message
[17:49:38] [main/WARN]: Reference map 'iris.refmap.json' for mixins.iris.vertexformat.json could not be read. If this is a development environment you can ignore this message
[17:49:38] [main/WARN]: Reference map 'iris.refmap.json' for mixins.iris.bettermipmaps.json could not be read. If this is a development environment you can ignore this message
[17:49:38] [main/WARN]: Reference map 'iris.refmap.json' for mixins.iris.compat.sodium.json could not be read. If this is a development environment you can ignore this message
[17:49:38] [main/WARN]: Reference map 'iris.refmap.json' for mixins.iris.fixes.maxfpscrash.json could not be read. If this is a development environment you can ignore this message
[17:49:38] [main/ERROR]: Failed reading REFMAP JSON from : java.lang.NullPointerException Cannot invoke "org.spongepowered.asm.mixin.refmap.ReferenceMapper.setResourceName(String)" because "mapper" is null
[17:49:38] [main/WARN]: Reference map '' for moderner_beta-common.mixins.json could not be read. If this is a development environment you can ignore this message
[17:49:38] [main/WARN]: Reference map 'moreculling.refmap.json' for moreculling.mixins.json could not be read. If this is a development environment you can ignore this message
[17:49:38] [main/WARN]: Reference map 'moreculling.refmap.json' for moreculling.fabric.mixins.json could not be read. If this is a development environment you can ignore this message
[17:49:38] [main/INFO]: Loaded configuration file for Sodium: 35 options available, 1 override(s) found
[17:49:38] [main/WARN]: Reference map 'TaxFreeLevels-common-refmap.json' for taxfreelevels-common-fabric.mixins.json could not be read. If this is a development environment you can ignore this message
[17:49:38] [main/WARN]: Error loading class: xyz/nucleoid/server/translations/impl/language/LanguageReader (java.lang.ClassNotFoundException: xyz/nucleoid/server/translations/impl/language/LanguageReader)
[17:49:38] [main/WARN]: @Mixin target xyz.nucleoid.server.translations.impl.language.LanguageReader was not found owo.mixins.json:extension.json5.LanguageReaderMixin from mod owo
[17:49:38] [main/WARN]: Error loading class: xyz/nucleoid/server/translations/impl/language/SystemDelegatedLanguage (java.lang.ClassNotFoundException: xyz/nucleoid/server/translations/impl/language/SystemDelegatedLanguage)
[17:49:38] [main/WARN]: Error loading class: traben/entity_model_features/models/parts/EMFModelPart (java.lang.ClassNotFoundException: traben/entity_model_features/models/parts/EMFModelPart)
[17:49:38] [main/WARN]: Force-disabling mixin 'features.render.world.sky.LevelRendererMixin' as rule 'mixin.features.render.world.sky' (added by mods [iris]) disables it and children
[17:49:38] [main/WARN]: Error loading class: com/moulberry/flashback/record/FlashbackMeta (java.lang.ClassNotFoundException: com/moulberry/flashback/record/FlashbackMeta)
[17:49:38] [main/WARN]: @Mixin target com.moulberry.flashback.record.FlashbackMeta was not found client.voxy.mixins.json:flashback.MixinFlashbackMeta from mod voxy
[17:49:38] [main/WARN]: Error loading class: com/moulberry/flashback/record/Recorder (java.lang.ClassNotFoundException: com/moulberry/flashback/record/Recorder)
[17:49:38] [main/WARN]: @Mixin target com.moulberry.flashback.record.Recorder was not found client.voxy.mixins.json:flashback.MixinFlashbackRecorder from mod voxy
[17:49:38] [main/WARN]: Error loading class: me/cortex/nvidium/RenderPipeline (java.lang.ClassNotFoundException: me/cortex/nvidium/RenderPipeline)
[17:49:38] [main/WARN]: @Mixin target me.cortex.nvidium.RenderPipeline was not found client.voxy.mixins.json:nvidium.MixinRenderPipeline from mod voxy
[17:49:38] [main/WARN]: Error loading class: org/popcraft/chunky/platform/FabricWorld (java.lang.ClassNotFoundException: org/popcraft/chunky/platform/FabricWorld)
[17:49:38] [main/WARN]: @Mixin target org.popcraft.chunky.platform.FabricWorld was not found common.voxy.mixins.json:chunky.MixinFabricWorld from mod voxy
[17:49:38] [main/INFO]: Searching for graphics cards...
[17:49:39] [main/INFO]: Found graphics adapter: LinuxPciAdapterInfo[vendor=NVIDIA, name=GB206 [GeForce RTX 5060], pciVendorId=0x10de, pciDeviceId=0x2d05]
[17:49:39] [main/INFO]: Found graphics adapter: LinuxPciAdapterInfo[vendor=AMD, name=Cezanne [Radeon Vega Series / Radeon Vega Mobile Series], pciVendorId=0x1002, pciDeviceId=0x1638]
[17:49:39] [main/WARN]: Sodium has applied one or more workarounds to prevent crashes or other issues on your system: [NVIDIA_THREADED_OPTIMIZATIONS_BROKEN, NO_ERROR_CONTEXT_UNSUPPORTED, AMD_GAME_OPTIMIZATION_BROKEN]
[17:49:39] [main/WARN]: This is not necessarily an issue, but it may result in certain features or optimizations being disabled. You can sometimes fix these issues by upgrading your graphics driver.
[17:49:39] [main/INFO]: Initializing MixinExtras via com.llamalad7.mixinextras.service.MixinExtrasServiceImpl(version=0.5.0).
[17:49:39] [main/INFO]: trender.mixins.json:client.ScreenAccessor from mod trender->@Invoker[METHOD_PROXY]::libgui$defaultHandleGameClickEvent(Lnet/minecraft/class_2558;Lnet/minecraft/class_310;Lnet/minecraft/class_437;)V should be static as its target is
[17:49:39] [Datafixer Bootstrap/INFO]: 287 Datafixer optimizations took 258 milliseconds
[17:49:43] [Render thread/INFO]: Environment: Environment[sessionHost=https://sessionserver.mojang.com, servicesHost=https://api.minecraftservices.com, profilesHost=https://api.mojang.com, name=PROD]
[17:49:43] [Render thread/INFO]: Setting user: MixedVictor
[17:49:43] [Render thread/INFO]: Initializing BedrockIfy.
[17:49:44] [Render thread/INFO]: Constructing components for combatnouveau:common
[17:49:44] [Render thread/INFO]: Dispatching loading event for config combatnouveau-common.toml
[17:49:44] [Render thread/INFO]: Dispatching loading event for config combatnouveau-client.toml
[17:49:44] [Render thread/INFO]: Constructing components for leavesbegone:common
[17:49:44] [Render thread/INFO]: [Moderner Beta] Initializing Moderner Beta...
[17:49:44] [Render thread/INFO]: Blodhgarm: Did you know it took four switcheroos to implement MatrixStackTransformer in the correct package?
[17:49:44] [Render thread/INFO]: Constructing components for puzzleslib:common
[17:49:44] [Render thread/INFO]: [me.cx.vy.cn.cg.Serialization]: Registered Config as RocksDB for config type StorageConfig
[17:49:44] [Render thread/INFO]: [me.cx.vy.cn.cg.Serialization]: Registered Config as Redis for config type StorageConfig
[17:49:44] [Render thread/INFO]: [me.cx.vy.cn.cg.Serialization]: Registered Config as ReadonlyCachingLayer for config type StorageConfig
[17:49:44] [Render thread/INFO]: [me.cx.vy.cn.cg.Serialization]: Registered Config2 as AutoFragmentationAdaptor for config type StorageConfig
[17:49:44] [Render thread/INFO]: [me.cx.vy.cn.cg.Serialization]: Registered Config as FragmentationAdaptor for config type StorageConfig
[17:49:44] [Render thread/INFO]: [me.cx.vy.cn.cg.Serialization]: Registered ConditionalStorageBackendConfig as ConditionalConfig for config type StorageConfig
[17:49:44] [Render thread/INFO]: [me.cx.vy.cn.cg.Serialization]: Registered Config as CompressionAdaptor for config type StorageConfig
[17:49:44] [Render thread/INFO]: [me.cx.vy.cn.cg.Serialization]: Registered BasicPathInsertionConfig as BasicPathConfig for config type StorageConfig
[17:49:44] [Render thread/INFO]: [me.cx.vy.cn.cg.Serialization]: Registered Config as LMDB for config type StorageConfig
[17:49:44] [Render thread/INFO]: [me.cx.vy.cn.cg.Serialization]: Registered Config as Memory for config type StorageConfig
[17:49:44] [Render thread/INFO]: [me.cx.vy.cn.cg.Serialization]: Registered Config as Serializer for config type SectionStorageConfig
[17:49:44] [Render thread/INFO]: [me.cx.vy.cn.cg.Serialization]: Registered Config as ZSTD for config type CompressorConfig
[17:49:44] [Render thread/INFO]: [me.cx.vy.cn.cg.Serialization]: Registered Config as LZ4 for config type CompressorConfig
[17:49:44] [Render thread/INFO]: [me.cx.vy.cn.cg.Serialization]: Registered 13 config types
[17:49:44] [Render thread/INFO]: voxy world gen v2 initializing
[17:49:44] [Render thread/INFO]: voxy networking initialized
[17:49:44] [Render thread/INFO]: Initializing BedrockIfy Client.
[17:49:44] [Render thread/INFO]: Initialized BedrockIfy Client
[17:49:44] [Render thread/INFO]: Constructing components for combatnouveau:client
[17:49:44] [Render thread/INFO]: [Indigo] Different rendering plugin detected; not applying Indigo.
[17:49:44] [Render thread/INFO]: Constructing components for leavesbegone:client
[17:49:44] [Render thread/INFO]: Checking mod updates...
[17:49:45] [Render thread/INFO]: Constructing components for puzzleslib:client
[17:49:45] [Render thread/INFO]: [STDOUT]: [LibGui] Initializing Client...
[17:49:45] [Render thread/INFO]: initializing voxy world gen v2 client
[17:49:45] [Render thread/INFO]: Backend library: LWJGL version 3.3.3-snapshot
[17:49:45] [Render thread/INFO]: Modifying process environment to apply workarounds for the NVIDIA graphics driver...
[17:49:45] [ForkJoinPool.commonPool-worker-1/WARN]: [Iris Update Check] This version doesn't have an update index, skipping.
[17:49:46] [Render thread/INFO]: Debug functionality is disabled.
[17:49:46] [Render thread/INFO]: ARB_direct_state_access detected, enabling DSA.
[17:49:46] [Render thread/ERROR]: Unrecognized clouds setting: true
[17:49:46] [Render thread/WARN]: Unable to resolve shader pack option menu element "VC_ATTENUATION" defined in shaders.properties
[17:49:46] [Render thread/WARN]: Unable to resolve shader pack option menu element "FIREFLIES" defined in shaders.properties
[17:49:46] [Render thread/WARN]: Unable to resolve shader pack option menu element "FIREFLIES_BRIGHTNESS" defined in shaders.properties
[17:49:46] [Render thread/INFO]: Profile: Custom (+21 options changed by user)
[17:49:46] [Render thread/INFO]: [me.cx.vy.ct.is.IrisShaderPatch]: External opaque shader patch applied
[17:49:46] [Render thread/INFO]: [me.cx.vy.ct.is.IrisShaderPatch]: External translucent shader patch applied
[17:49:46] [Render thread/INFO]: Using shaderpack: Solas Shader V3.3.zip
[17:49:46] [Render thread/INFO]: OpenGL Vendor: NVIDIA Corporation
[17:49:46] [Render thread/INFO]: OpenGL Renderer: NVIDIA GeForce RTX 5060/PCIe/SSE2
[17:49:46] [Render thread/INFO]: OpenGL Version: 3.3.0 NVIDIA 580.126.09
[17:49:46] [Render thread/INFO]: Modifying OpenGL context to apply workarounds for the NVIDIA graphics driver...
[17:49:46] [Render thread/INFO]: Using optional rendering extensions: GL_ARB_buffer_storage, GL_KHR_debug, GL_ARB_vertex_attrib_binding, GL_ARB_direct_state_access, GL_EXT_texture_filter_anisotropic
[17:49:46] [Render thread/ERROR]: Couldn't load file/Iris Button Everywhere.1.6.zip pack metadata: Pack declares support for version newer than 64, but is missing mandatory fields min_format and max_format
[17:49:46] [Render thread/INFO]: Found non-pack entry '/home/********/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/instances/MixedPack/minecraft/resourcepacks/.index', ignoring
[W][17:49:46.838980] pw.conf      | [          conf.c: 1204 pw_conf_load_conf_for_context()] setting config.name to client-rt.conf is deprecated, using client.conf
[ALSOFT] (EE) Failed to connect PipeWire event context (errno: 112)
[17:49:46] [Render thread/INFO]: Block minecraft:bubble_column had its color provider replaced with mod.bluestaggo.modernerbeta.client.color.BlockColors$$Lambda/0x00000008011cfbe0@192d059c and will not use per-vertex coloring
[17:49:46] [Render thread/INFO]: Block minecraft:pink_petals had its color provider replaced with mod.bluestaggo.modernerbeta.client.color.BlockColors$$Lambda/0x00000008011cf598@4eef1c52 and will not use per-vertex coloring
[17:49:46] [Render thread/INFO]: Block minecraft:grass_block had its color provider replaced with mod.bluestaggo.modernerbeta.client.color.BlockColors$$Lambda/0x00000008011cef50@ad30ea0 and will not use per-vertex coloring
[17:49:46] [Render thread/INFO]: Block minecraft:vine had its color provider replaced with mod.bluestaggo.modernerbeta.client.color.BlockColors$$Lambda/0x00000008011cf7b0@7d77e509 and will not use per-vertex coloring
[17:49:46] [Render thread/INFO]: Block minecraft:wildflowers had its color provider replaced with mod.bluestaggo.modernerbeta.client.color.BlockColors$$Lambda/0x00000008011cf598@4eef1c52 and will not use per-vertex coloring
[17:49:46] [Render thread/INFO]: Block minecraft:short_grass had its color provider replaced with mod.bluestaggo.modernerbeta.client.color.BlockColors$$Lambda/0x00000008011cf168@c071ede and will not use per-vertex coloring
[17:49:46] [Render thread/INFO]: Block minecraft:potted_fern had its color provider replaced with mod.bluestaggo.modernerbeta.client.color.BlockColors$$Lambda/0x00000008011cf168@c071ede and will not use per-vertex coloring
[17:49:46] [Render thread/INFO]: Block minecraft:bush had its color provider replaced with mod.bluestaggo.modernerbeta.client.color.BlockColors$$Lambda/0x00000008011cf168@c071ede and will not use per-vertex coloring
[17:49:46] [Render thread/INFO]: Block minecraft:jungle_leaves had its color provider replaced with mod.bluestaggo.modernerbeta.client.color.BlockColors$$Lambda/0x00000008011cf7b0@7d77e509 and will not use per-vertex coloring
[17:49:46] [Render thread/INFO]: Block minecraft:fern had its color provider replaced with mod.bluestaggo.modernerbeta.client.color.BlockColors$$Lambda/0x00000008011cf168@c071ede and will not use per-vertex coloring
[17:49:46] [Render thread/INFO]: Block minecraft:sugar_cane had its color provider replaced with mod.bluestaggo.modernerbeta.client.color.BlockColors$$Lambda/0x00000008011cf9c8@3abc8e80 and will not use per-vertex coloring
[17:49:46] [Render thread/INFO]: Block minecraft:water had its color provider replaced with mod.bluestaggo.modernerbeta.client.color.BlockColors$$Lambda/0x00000008011cfbe0@192d059c and will not use per-vertex coloring
[17:49:46] [Render thread/INFO]: Block minecraft:tall_grass had its color provider replaced with mod.bluestaggo.modernerbeta.client.color.BlockColors$$Lambda/0x00000008011cf380@28882179 and will not use per-vertex coloring
[17:49:46] [Render thread/INFO]: Block minecraft:water_cauldron had its color provider replaced with mod.bluestaggo.modernerbeta.client.color.BlockColors$$Lambda/0x00000008011cfbe0@192d059c and will not use per-vertex coloring
[17:49:46] [Render thread/INFO]: Block minecraft:mangrove_leaves had its color provider replaced with mod.bluestaggo.modernerbeta.client.color.BlockColors$$Lambda/0x00000008011cf7b0@7d77e509 and will not use per-vertex coloring
[17:49:46] [Render thread/INFO]: Block minecraft:large_fern had its color provider replaced with mod.bluestaggo.modernerbeta.client.color.BlockColors$$Lambda/0x00000008011cf380@28882179 and will not use per-vertex coloring
[17:49:46] [Render thread/INFO]: Block minecraft:oak_leaves had its color provider replaced with mod.bluestaggo.modernerbeta.client.color.BlockColors$$Lambda/0x00000008011cf7b0@7d77e509 and will not use per-vertex coloring
[17:49:46] [Render thread/INFO]: Block minecraft:acacia_leaves had its color provider replaced with mod.bluestaggo.modernerbeta.client.color.BlockColors$$Lambda/0x00000008011cf7b0@7d77e509 and will not use per-vertex coloring
[17:49:46] [Render thread/INFO]: Block minecraft:dark_oak_leaves had its color provider replaced with mod.bluestaggo.modernerbeta.client.color.BlockColors$$Lambda/0x00000008011cf7b0@7d77e509 and will not use per-vertex coloring
[17:49:47] [Render thread/INFO]: Hardware information:
[17:49:47] [Render thread/INFO]: CPU: 16x AMD Ryzen 7 5700G with Radeon Graphics
[17:49:47] [Render thread/INFO]: GPU: NVIDIA GeForce RTX 5060/PCIe/SSE2 (Supports OpenGL 3.3.0 NVIDIA 580.126.09)
[17:49:47] [Render thread/INFO]: OS: Linux (6.17.0-14-generic)
[17:49:47] [Render thread/INFO]: Reloading ResourceManager: vanilla, bedrockify, cloth-config, combatnouveau, conditional-mixin, continuity, cubes-without-borders, fabric-api, fabric-api-base, fabric-api-lookup-api-v1, fabric-biome-api-v1, fabric-block-api-v1, fabric-block-view-api-v2, fabric-command-api-v2, fabric-content-registries-v0, fabric-convention-tags-v1, fabric-convention-tags-v2, fabric-crash-report-info-v1, fabric-data-attachment-api-v1, fabric-data-generation-api-v1, fabric-dimensions-v1, fabric-entity-events-v1, fabric-events-interaction-v0, fabric-game-rule-api-v1, fabric-item-api-v1, fabric-item-group-api-v1, fabric-key-binding-api-v1, fabric-language-kotlin, fabric-lifecycle-events-v1, fabric-loot-api-v2, fabric-loot-api-v3, fabric-message-api-v1, fabric-model-loading-api-v1, fabric-networking-api-v1, fabric-object-builder-api-v1, fabric-particles-v1, fabric-recipe-api-v1, fabric-registry-sync-v0, fabric-renderer-api-v1, fabric-renderer-indigo, fabric-rendering-fluids-v1, fabric-rendering-v1, fabric-resource-conditions-api-v1, fabric-resource-loader-v0, fabric-resource-loader-v1, fabric-screen-api-v1, fabric-screen-handler-api-v1, fabric-serialization-api-v1, fabric-sound-api-v1, fabric-tag-api-v1, fabric-transfer-api-v1, fabric-transitive-access-wideners-v1, fabricloader, forgeconfigapiport, hardcover, hunger_remover, iris, lattice, midnightlib, moderner_beta, modmenu, moreculling, owo, placeholder-api, skinlayers3d, smallviewmodel, sodium, taxfreelevels, trender, voxy, voxyworldgenv2, programmer_art, fabric-item-group-api-v1_programmer_art, modmenu_programmer_art, file/PACP_Base_Release.zip, file/golden-days-base-1.21.x-1.14.0.zip, file/Iris Button Everywhere.1.6.zip, file/Myth's Better Hotbar.zip, file/Shiny Hearts 1.1.zip, file/Aimz - Better Crosshair.zip
[17:49:47] [Render thread/INFO]: Cached all modded block culling states
[17:49:47] [Worker-Main-7/WARN]: Invalid sounds.json in resourcepack: 'file/PACP_Base_Release.zip'
com.google.gson.JsonParseException: com.google.gson.stream.MalformedJsonException: Expected value at line 1 column 3 path $
See https://github.com/google/gson/blob/main/Troubleshooting.md#malformed-json
	at knot//net.minecraft.class_3518.method_47992(class_3518.java:532)
	at knot//net.minecraft.class_3518.method_15297(class_3518.java:537)
	at knot//net.minecraft.class_1144.method_18180(class_1144.java:80)
	at knot//net.minecraft.class_1144.method_18789(class_1144.java:41)
	at knot//net.minecraft.class_4080.method_18791(class_4080.java:13)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1768)
	at knot//net.minecraft.class_4014.method_67576(class_4014.java:51)
	at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1423)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:387)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1312)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1843)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1808)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:188)
Caused by: com.google.gson.stream.MalformedJsonException: Expected value at line 1 column 3 path $
See https://github.com/google/gson/blob/main/Troubleshooting.md#malformed-json
	at knot//com.google.gson.stream.JsonReader.syntaxError(JsonReader.java:1817)
	at knot//com.google.gson.stream.JsonReader.doPeek(JsonReader.java:726)
	at knot//com.google.gson.stream.JsonReader.peek(JsonReader.java:547)
	at knot//com.google.gson.internal.bind.MapTypeAdapterFactory$Adapter.read(MapTypeAdapterFactory.java:177)
	at knot//com.google.gson.internal.bind.MapTypeAdapterFactory$Adapter.read(MapTypeAdapterFactory.java:161)
	at knot//net.minecraft.class_3518.method_47992(class_3518.java:530)
	... 12 more
[17:49:47] [Worker-Main-7/WARN]: File minecraft:sounds/music/game/creative/creative1.ogg does not exist, cannot add it to event minecraft:music.creative
[17:49:47] [Worker-Main-7/WARN]: File minecraft:sounds/music/game/creative/creative2.ogg does not exist, cannot add it to event minecraft:music.creative
[17:49:47] [Worker-Main-7/WARN]: File minecraft:sounds/music/game/creative/creative3.ogg does not exist, cannot add it to event minecraft:music.creative
[17:49:47] [Worker-Main-7/WARN]: File minecraft:sounds/music/game/creative/creative4.ogg does not exist, cannot add it to event minecraft:music.creative
[17:49:47] [Worker-Main-7/WARN]: File minecraft:sounds/music/game/creative/creative5.ogg does not exist, cannot add it to event minecraft:music.creative
[17:49:47] [Worker-Main-7/WARN]: File minecraft:sounds/music/game/creative/creative6.ogg does not exist, cannot add it to event minecraft:music.creative
[17:49:47] [Worker-Main-7/WARN]: File minecraft:sounds/music/game/calm1.ogg does not exist, cannot add it to event minecraft:music.game
[17:49:47] [Worker-Main-7/WARN]: File minecraft:sounds/music/game/calm2.ogg does not exist, cannot add it to event minecraft:music.game
[17:49:47] [Worker-Main-7/WARN]: File minecraft:sounds/music/game/calm3.ogg does not exist, cannot add it to event minecraft:music.game
[17:49:47] [Worker-Main-7/WARN]: File minecraft:sounds/music/game/hal1.ogg does not exist, cannot add it to event minecraft:music.game
[17:49:47] [Worker-Main-7/WARN]: File minecraft:sounds/music/game/hal2.ogg does not exist, cannot add it to event minecraft:music.game
[17:49:47] [Worker-Main-7/WARN]: File minecraft:sounds/music/game/hal3.ogg does not exist, cannot add it to event minecraft:music.game
[17:49:47] [Worker-Main-7/WARN]: File minecraft:sounds/music/game/hal4.ogg does not exist, cannot add it to event minecraft:music.game
[17:49:47] [Worker-Main-7/WARN]: File minecraft:sounds/music/game/nuance1.ogg does not exist, cannot add it to event minecraft:music.game
[17:49:47] [Worker-Main-7/WARN]: File minecraft:sounds/music/game/nuance2.ogg does not exist, cannot add it to event minecraft:music.game
[17:49:47] [Worker-Main-7/WARN]: File minecraft:sounds/music/game/piano1.ogg does not exist, cannot add it to event minecraft:music.game
[17:49:47] [Worker-Main-7/WARN]: File minecraft:sounds/music/game/piano2.ogg does not exist, cannot add it to event minecraft:music.game
[17:49:47] [Worker-Main-7/WARN]: File minecraft:sounds/music/game/piano3.ogg does not exist, cannot add it to event minecraft:music.game
[17:49:47] [Worker-Main-7/WARN]: File minecraft:sounds/music/menu/menu1.ogg does not exist, cannot add it to event minecraft:music.menu
[17:49:47] [Worker-Main-7/WARN]: File minecraft:sounds/music/menu/menu2.ogg does not exist, cannot add it to event minecraft:music.menu
[17:49:47] [Worker-Main-7/WARN]: File minecraft:sounds/music/menu/menu3.ogg does not exist, cannot add it to event minecraft:music.menu
[17:49:47] [Worker-Main-7/WARN]: File minecraft:sounds/music/menu/menu4.ogg does not exist, cannot add it to event minecraft:music.menu
[17:49:47] [Worker-Main-7/WARN]: File minecraft:sounds/music/game/nether/nether1.ogg does not exist, cannot add it to event minecraft:music.nether
[17:49:47] [Worker-Main-7/WARN]: File minecraft:sounds/music/game/nether/nether2.ogg does not exist, cannot add it to event minecraft:music.nether
[17:49:47] [Worker-Main-7/WARN]: File minecraft:sounds/music/game/nether/nether3.ogg does not exist, cannot add it to event minecraft:music.nether
[17:49:47] [Worker-Main-7/WARN]: File minecraft:sounds/music/game/nether/nether4.ogg does not exist, cannot add it to event minecraft:music.nether
[17:49:47] [Worker-Main-16/INFO]: Found unifont_all_no_pua-17.0.01.hex, loading
[17:49:47] [Worker-Main-16/INFO]: Found unifont_jp_patch-17.0.01.hex, loading
[17:49:47] [Worker-Main-16/INFO]: Found unifont_pua-17.0.01.hex, loading
[17:49:49] [Worker-Main-8/WARN]: Missing block model: minecraft:block/piston_top
[17:49:49] [Worker-Main-8/WARN]: Missing block model: minecraft:builtin/entity
[17:49:49] [Worker-Main-12/WARN]: Unresolved texture references in minecraft:block/lectern:
	#west-> #side
	#down-> #bottom
	#south-> #side
	#east-> #side
	#north-> #front
	#up-> #top
	#bottom-> #top
[17:49:49] [Worker-Main-14/WARN]: Unresolved texture references in minecraft:block/loom:
	#bottom-> #top
[17:49:49] [Worker-Main-6/WARN]: Unresolved texture references in minecraft:block/end_portal_frame_filled:
	#west-> #all
	#down-> #all
	#south-> #all
	#east-> #all
	#north-> #all
	#up-> #all
[17:49:49] [Worker-Main-10/WARN]: Unresolved texture references in minecraft:item/template_pressure_plate:
	#particle-> #texture
[17:49:49] [Worker-Main-15/WARN]: Missing textures in model minecraft:item/decorated_pot:
    minecraft:block_or_item:minecraft:entity/decorated_pot/decorated_pot_side
[17:49:49] [Worker-Main-15/WARN]: Missing texture references in model minecraft:item/template_pressure_plate:
    #texture
    particle
[17:49:49] [Render thread/INFO]: OpenAL initialized on device Renoir/Cezanne HDMI/DP Audio Controller Digital Stereo (HDMI)
[17:49:49] [Render thread/INFO]: Sound engine started
[17:49:49] [Render thread/INFO]: Created: 512x256x0 minecraft:textures/atlas/particles.png-atlas
[17:49:49] [Render thread/INFO]: Created: 128x128x0 minecraft:textures/atlas/decorated_pot.png-atlas
[17:49:49] [Render thread/INFO]: Created: 2048x1024x0 minecraft:textures/atlas/armor_trims.png-atlas
[17:49:49] [Render thread/INFO]: Created: 512x256x0 minecraft:textures/atlas/paintings.png-atlas
[17:49:49] [Render thread/INFO]: Created: 1024x512x0 minecraft:textures/atlas/shield_patterns.png-atlas
[17:49:49] [Render thread/INFO]: Created: 2048x2048x4 minecraft:textures/atlas/blocks.png-atlas
[17:49:49] [Render thread/INFO]: Created: 512x512x0 minecraft:textures/atlas/chest.png-atlas
[17:49:49] [Render thread/INFO]: Created: 256x128x0 minecraft:textures/atlas/celestials.png-atlas
[17:49:49] [Render thread/INFO]: Created: 1024x512x0 minecraft:textures/atlas/banner_patterns.png-atlas
[17:49:49] [Render thread/INFO]: Created: 512x512x0 minecraft:textures/atlas/beds.png-atlas
[17:49:49] [Render thread/INFO]: Created: 1024x512x0 minecraft:textures/atlas/items.png-atlas
[17:49:49] [Render thread/INFO]: Created: 1024x1024x0 minecraft:textures/atlas/gui.png-atlas
[17:49:49] [Render thread/INFO]: Created: 128x64x0 minecraft:textures/atlas/map_decorations.png-atlas
[17:49:49] [Render thread/INFO]: Created: 512x256x0 minecraft:textures/atlas/signs.png-atlas
[17:49:49] [Render thread/INFO]: Created: 512x512x0 minecraft:textures/atlas/shulker_boxes.png-atlas
[17:49:50] [Render thread/INFO]: 29689 cache(s) where cleared!
[17:49:50] [Render thread/INFO]: Creating pipeline for dimension minecraft:overworld
[17:49:51] [Render thread/INFO]: [Bedrockify] The condition matches. BedrockIfy Sun Glare Shading is now disabled.
[17:50:03] [Render thread/INFO]: Loaded 1555 recipes
[17:50:03] [Render thread/INFO]: Loaded 1584 advancements
[17:50:03] [Render thread/INFO]: Applied 16 biome modifications to 11 of 121 new biomes in 4.113 ms
[17:50:03] [Server thread/INFO]: Dispatching loading event for config combatnouveau-server.toml
[17:50:03] [Server thread/INFO]: Dispatching loading event for config leavesbegone-server.toml
[17:50:03] [Server thread/INFO]: Starting integrated minecraft server version 1.21.11
[17:50:03] [Server thread/INFO]: Generating keypair
[17:50:04] [Server thread/INFO]: Loading 80 persistent chunks...
[17:50:04] [Server thread/INFO]: Preparing spawn area: 7%
[17:50:04] [Server thread/INFO]: Preparing spawn area: 7%
[17:50:05] [Server thread/INFO]: Preparing spawn area: 7%
[17:50:05] [Server thread/INFO]: Preparing spawn area: 7%
[17:50:06] [Server thread/INFO]: Preparing spawn area: 7%
[17:50:06] [Server thread/INFO]: server started, initializing manager
[17:50:06] [Server thread/INFO]: voxy world gen initialized
[17:50:07] [Server thread/INFO]: Loading 49 chunks for player spawn...
[17:50:07] [Server thread/INFO]: Preparing spawn area: 64%
[17:50:07] [Server thread/INFO]: Preparing spawn area: 100%
[17:50:07] [Server thread/INFO]: Time elapsed: 3007 ms
[17:50:07] [Server thread/INFO]: MixedVictor[local:E:305d161a] logged in with entity id 93 at (457.26832623471904, 65.0, -50.915595465699795)
[17:50:07] [Server thread/INFO]: Sending server config values to client
[17:50:07] [Render thread/INFO]: [me.cx.vy.cl.VoxyInstance]: Initializing voxy instance
[17:50:07] [Server thread/INFO]: MixedVictor joined the game
[17:50:07] [Render thread/INFO]: [me.cx.vy.cl.VoxyInstance]: Dedicated voxy thread pool size: 6
[17:50:07] [Render thread/INFO]: Reloading pipeline on dimension change: minecraft:overworld => minecraft:overworld
[17:50:07] [Render thread/INFO]: Destroying pipeline minecraft:overworld
[17:50:07] [Render thread/INFO]: Creating pipeline for dimension minecraft:overworld
[17:50:07] [Server thread/WARN]: tellus not found for sampling integration: com.yucareux.tellus.worldgen.EarthChunkGenerator
[17:50:07] [Server thread/INFO]: voxy ingester initialized successfully (reflective)
[17:50:07] [Server thread/INFO]: tellus integration hub initialized (8 workers)
[17:50:07] [Server thread/INFO]: loaded 31104 chunks from voxy generation cache for ResourceKey[minecraft:dimension / minecraft:overworld]
[17:50:07] [Server thread/INFO]: Changing view distance to 16, from 10
[17:50:07] [Server thread/INFO]: Changing simulation distance to 16, from 0
[17:50:07] [Render thread/INFO]: [me.cx.vy.cl.VoxyInstance]: Creating new world engine: 3477408326914914500@1317752649
[17:50:07] [Render thread/INFO]: [me.cx.vy.ct.ce.rg.sn.gy.BasicSectionGeometryData]: Creating and zeroing 4095MB geometry buffer driver states 5978MB of free memory
[17:50:07] [Render thread/INFO]: [me.cx.vy.ct.ce.rg.sn.gy.BasicSectionGeometryData]: if your game crashes/exits here without any other log message, try manually decreasing the geometry capacity
[17:50:08] [Render thread/INFO]: [me.cx.vy.ct.ce.rg.sn.gy.BasicSectionGeometryData]: Successfully allocated the geometry buffer in 7ms
[17:50:08] [Render thread/INFO]: [me.cx.vy.ct.ce.RenderPipelineFactory]: Creating voxy iris render pipeline
[17:50:08] [Render thread/INFO]: [me.cx.vy.ct.ce.VoxyRenderSystem]: Voxy render system created with 4294966272 geometry capacity, using pipeline 'IrisVoxyRenderPipeline' with renderer 'MDICSectionRenderer'
[17:50:08] [Render thread/INFO]: Started 10 worker threads
[17:50:08] [Render thread/INFO]: [me.cx.vy.cl.VoxyInstance]: Dedicated voxy thread pool size: 1
[17:50:08] [Render thread/WARN]: [me.cx.vy.cl.is.DHImporter]: Unable to load sqlite JDBC or lzma decompressor, DHImporting wont be available
[17:50:08] [Render thread/INFO]: Applying server overrides
[17:50:08] [Render thread/INFO]: Responding with client values
[17:50:08] [Render thread/INFO]: Loaded 14 advancements
[17:50:08] [Server thread/INFO]: Receiving client config
[17:50:08] [Render thread/INFO]: [me.cx.vy.ct.ce.VoxyRenderSystem]: Flushing download stream
[17:50:08] [Render thread/INFO]: [me.cx.vy.ct.ce.VoxyRenderSystem]: Shutting down rendering
[17:50:08] [Render thread/INFO]: [me.cx.vy.ct.ce.rg.sn.gy.BasicSectionGeometryData]: Attempting to wait for gpu memory to release
[17:50:08] [Render thread/WARN]: [me.cx.vy.ct.ce.rg.sn.gy.BasicSectionGeometryData]: Failed to wait for gpu memory to be freed, this could indicate an issue with the driver
[17:50:08] [Render thread/INFO]: [me.cx.vy.ct.ce.VoxyRenderSystem]: Shutting down render pipeline
[17:50:08] [Render thread/INFO]: [me.cx.vy.ct.ce.VoxyRenderSystem]: Flushing download stream
[17:50:08] [Render thread/INFO]: [me.cx.vy.ct.ce.VoxyRenderSystem]: Render shutdown completed
[17:50:09] [Render thread/INFO]: [me.cx.vy.ct.ce.rg.sn.gy.BasicSectionGeometryData]: Creating and zeroing 4095MB geometry buffer driver states 5975MB of free memory
[17:50:09] [Render thread/INFO]: [me.cx.vy.ct.ce.rg.sn.gy.BasicSectionGeometryData]: if your game crashes/exits here without any other log message, try manually decreasing the geometry capacity
[17:50:09] [Render thread/INFO]: [me.cx.vy.ct.ce.rg.sn.gy.BasicSectionGeometryData]: Successfully allocated the geometry buffer in 6ms
[17:50:09] [Render thread/INFO]: [me.cx.vy.ct.ce.RenderPipelineFactory]: Creating voxy iris render pipeline
[17:50:09] [Render thread/INFO]: [me.cx.vy.ct.ce.VoxyRenderSystem]: Voxy render system created with 4294966272 geometry capacity, using pipeline 'IrisVoxyRenderPipeline' with renderer 'MDICSectionRenderer'
[17:50:09] [Render thread/INFO]: Stopping worker threads
[17:50:09] [Render thread/INFO]: Started 10 worker threads
[17:50:09] [Render thread/INFO]: Resizing Dynamic Transforms UBO, capacity limit of 2 reached during a single frame. New capacity will be 4.
[17:50:09] [Render thread/INFO]: Resizing Dynamic Transforms UBO, capacity limit of 4 reached during a single frame. New capacity will be 8.
[17:50:09] [Worker-Main-10/WARN]: Empty height range: [136 absolute-0 below top]
[17:50:09] [Render thread/INFO]: Resizing Dynamic Transforms UBO, capacity limit of 8 reached during a single frame. New capacity will be 16.
[17:50:12] [Render thread/INFO]: voxy integration initialized (enabled: true, raw: true)
[17:50:16] [Async Node Manager/INFO]: [me.cx.vy.ct.ce.rg.hl.AsyncNodeManager$ComputeMemoryCopy]: Resizing scratch data buffer to: 2097152
[17:50:20] [Async Node Manager/INFO]: [me.cx.vy.ct.ce.rg.hl.AsyncNodeManager$ComputeMemoryCopy]: Resizing scratch data buffer to: 4194304
[17:50:23] [Server thread/WARN]: Can't keep up! Is the server overloaded? Running 6581ms or 131 ticks behind
[17:50:25] [Async Node Manager/INFO]: [me.cx.vy.ct.ce.rg.hl.AsyncNodeManager$SyncResults]: Expanding scatter update buffer to 24720
[17:50:25] [Async Node Manager/INFO]: [me.cx.vy.ct.ce.rg.hl.AsyncNodeManager$ComputeMemoryCopy]: Resizing scratch data buffer to: 8388608
[17:50:28] [Async Node Manager/INFO]: [me.cx.vy.ct.ce.rg.hl.AsyncNodeManager$SyncResults]: Expanding scatter update buffer to 37200
[17:50:30] [Async Node Manager/INFO]: [me.cx.vy.ct.ce.rg.hl.AsyncNodeManager$ComputeMemoryCopy]: Resizing scratch data buffer to: 2097152
[17:50:30] [Async Node Manager/INFO]: [me.cx.vy.ct.ce.rg.hl.AsyncNodeManager$ComputeMemoryCopy]: Resizing scratch data buffer to: 4194304
[17:50:30] [Async Node Manager/INFO]: [me.cx.vy.ct.ce.rg.hl.AsyncNodeManager$ComputeMemoryCopy]: Resizing scratch data buffer to: 8388608
[17:50:30] [Async Node Manager/INFO]: [me.cx.vy.ct.ce.rg.hl.AsyncNodeManager$SyncResults]: Expanding scatter update buffer to 24720
[17:50:37] [Async Node Manager/INFO]: [me.cx.vy.ct.ce.rg.hl.AsyncNodeManager$SyncResults]: Expanding scatter update buffer to 37200
[17:51:39] [Render thread/INFO]: [System] [CHAT] Saved screenshot as 2026-02-16_17.51.37.png
Exception: java.lang.OutOfMemoryError thrown from the UncaughtExceptionHandler in thread "JNA Cleaner"
[17:52:54] [Server thread/ERROR]: Uncaught exception in server thread
java.lang.OutOfMemoryError: Java heap space
[17:52:54] [Active world cleaner/INFO]: [STDERR]: java.lang.OutOfMemoryError: Java heap space
[17:52:54] [HttpClient-1-SelectorManager/ERROR]: Uncaught exception in thread "HttpClient-1-SelectorManager"
java.lang.OutOfMemoryError: Java heap space
[17:52:54] [Active world cleaner/INFO]: [STDERR]: 	at java.base/java.util.HashMap$EntrySet.iterator(HashMap.java:1106)
[17:52:54] [Active world cleaner/INFO]: [STDERR]: 	at knot//me.cortex.voxy.commonImpl.VoxyInstance.cleanIdle(VoxyInstance.java:186)
[17:52:54] [Active world cleaner/INFO]: [STDERR]: 	at knot//me.cortex.voxy.commonImpl.VoxyInstance.lambda$new$1(VoxyInstance.java:46)
[17:52:54] [Active world cleaner/INFO]: [STDERR]: 	at knot//me.cortex.voxy.commonImpl.VoxyInstance$$Lambda/0x0000000801b73930.run(Unknown Source)
[17:52:54] [Active world cleaner/INFO]: [STDERR]: 	at java.base/java.lang.Thread.runWith(Thread.java:1596)
[17:52:54] [Active world cleaner/INFO]: [STDERR]: 	at java.base/java.lang.Thread.run(Thread.java:1583)
[17:52:54] [Render thread/INFO]: [me.cx.vy.ct.ce.VoxyRenderSystem]: Flushing download stream
[17:52:54] [Render thread/INFO]: [me.cx.vy.ct.ce.VoxyRenderSystem]: Shutting down rendering
[17:52:54] [Render thread/INFO]: [me.cx.vy.ct.ce.VoxyRenderSystem]: Shutting down render pipeline
[17:52:54] [Render thread/INFO]: [me.cx.vy.ct.ce.VoxyRenderSystem]: Flushing download stream
[17:52:54] [Render thread/INFO]: [me.cx.vy.ct.ce.VoxyRenderSystem]: Render shutdown completed
[17:52:54] [Render thread/INFO]: Stopping worker threads
[17:52:54] [Render thread/INFO]: [me.cx.vy.cl.VoxyInstance]: Shutting down voxy instance
Process crashed with exitcode 6.
Clipboard copy at: 16 Feb 2026 17:59:02 -0300
```

--- a/src/main/java/com/ethan/voxyworldgenv2/core/ChunkGenerationManager.java
+++ b/src/main/java/com/ethan/voxyworldgenv2/core/ChunkGenerationManager.java
@@ -203,6 +203,7 @@ public final class ChunkGenerationManager {
                             final List<ChunkPos> finalSyncBatch = new ArrayList<>(syncBatch);
                             final ServerLevel level = ds.level;
                             final UUID playerUUID = player.getUUID();
+                            
                             server.execute(() -> {
                                 ServerPlayer p = server.getPlayerList().getPlayer(playerUUID);
                                 if (p != null) {
@@ -210,6 +211,10 @@ public final class ChunkGenerationManager {
                                         LevelChunk c = level.getChunkSource().getChunk(syncPos.x, syncPos.z, false);
                                         if (c != null) {
                                             com.ethan.voxyworldgenv2.network.NetworkHandler.sendLODData(p, c);
+                                            synced.add(syncPos.toLong());
+                                        } else {
+                                            // 如果区块其实不在内存里，发送失败，回滚
+                                            synced.remove(syncPos.toLong());
                                         }
                                     }
                                 }
@@ -218,7 +223,10 @@ public final class ChunkGenerationManager {
                         }
                     }
                     
-                    if (workDispatched) continue; 
+                    if (workDispatched) {
+                        Thread.sleep(10); // small delay to prevent overwhelming network/server tasks
+                        continue; 
+                    }
                     
                     Thread.sleep(100);
                     continue;

--- a/src/main/java/com/ethan/voxyworldgenv2/core/ChunkGenerationManager.java
+++ b/src/main/java/com/ethan/voxyworldgenv2/core/ChunkGenerationManager.java
@@ -203,7 +203,6 @@ public final class ChunkGenerationManager {
                             final List<ChunkPos> finalSyncBatch = new ArrayList<>(syncBatch);
                             final ServerLevel level = ds.level;
                             final UUID playerUUID = player.getUUID();
-                            
                             server.execute(() -> {
                                 ServerPlayer p = server.getPlayerList().getPlayer(playerUUID);
                                 if (p != null) {
@@ -212,9 +211,6 @@ public final class ChunkGenerationManager {
                                         if (c != null) {
                                             com.ethan.voxyworldgenv2.network.NetworkHandler.sendLODData(p, c);
                                             synced.add(syncPos.toLong());
-                                        } else {
-                                            // 如果区块其实不在内存里，发送失败，回滚
-                                            synced.remove(syncPos.toLong());
                                         }
                                     }
                                 }

--- a/src/main/java/com/ethan/voxyworldgenv2/network/NetworkHandler.java
+++ b/src/main/java/com/ethan/voxyworldgenv2/network/NetworkHandler.java
@@ -92,6 +92,17 @@ public class NetworkHandler {
         VoxyWorldGenV2.LOGGER.info("voxy networking initialized");
     }
 
+    private static void setSyncedState(ServerPlayer player, ChunkPos pos, boolean isSynced) {
+        var synced = PlayerTracker.getInstance().getSyncedChunks(player.getUUID());
+        if (synced != null) {
+            if (isSynced) {
+                synced.add(pos.toLong());
+            } else {
+                synced.remove(pos.toLong());
+            }
+        }
+    }
+
     public static void broadcastLODData(LevelChunk chunk) {
         ChunkPos pos = chunk.getPos();
         int minY = chunk.getMinSectionY();
@@ -137,30 +148,38 @@ public class NetworkHandler {
         }
         
         if (sections.isEmpty()) return;
-        
+
         LODDataPayload payload = new LODDataPayload(pos, minY, sections);
         
         double maxDistSq = 4096.0 * 4096.0;
         
         for (ServerPlayer player : PlayerTracker.getInstance().getPlayers()) {
-            if (player.level() != chunk.getLevel()) continue;
-            
             double dx = player.getX() - (pos.getMiddleBlockX());
             double dz = player.getZ() - (pos.getMiddleBlockZ());
-            if (dx * dx + dz * dz <= maxDistSq) {
-                ServerPlayNetworking.send(player, payload);
-                
-                // mark as synced for this player
-                var synced = PlayerTracker.getInstance().getSyncedChunks(player.getUUID());
-                if (synced != null) {
-                    synced.add(pos.toLong());
-                }
+            
+            if (player.level() != chunk.getLevel() || (dx * dx + dz * dz > maxDistSq)) {
+                setSyncedState(player, pos, false);
+                continue;
             }
+            
+            ServerPlayNetworking.send(player, payload);
+                
+            // // mark as synced for this player
+            // var synced = PlayerTracker.getInstance().getSyncedChunks(player.getUUID());
+            // if (synced != null) {
+            //     synced.add(pos.toLong());
+            // }
         }
     }
 
     public static void sendLODData(ServerPlayer player, LevelChunk chunk) {
         ChunkPos pos = chunk.getPos();
+        
+        if (player.level() != chunk.getLevel()) {
+            setSyncedState(player, pos, false);
+            return;
+        }
+
         int minY = chunk.getMinSectionY();
         List<LODDataPayload.SectionData> sections = new ArrayList<>();
         
@@ -201,14 +220,14 @@ public class NetworkHandler {
             ));
         }
         
-        if (sections.isEmpty()) return;
+        if (sections.isEmpty()) {
+            setSyncedState(player, pos, false);
+            return;
+        }
         
         ServerPlayNetworking.send(player, new LODDataPayload(pos, minY, sections));
-        
-        var synced = PlayerTracker.getInstance().getSyncedChunks(player.getUUID());
-        if (synced != null) {
-            synced.add(pos.toLong());
-        }
+        // double check, could be deleted.
+        setSyncedState(player, pos, true);
     }
 
     public static void sendHandshake(ServerPlayer player) {

--- a/src/main/java/com/ethan/voxyworldgenv2/network/NetworkHandler.java
+++ b/src/main/java/com/ethan/voxyworldgenv2/network/NetworkHandler.java
@@ -174,12 +174,6 @@ public class NetworkHandler {
 
     public static void sendLODData(ServerPlayer player, LevelChunk chunk) {
         ChunkPos pos = chunk.getPos();
-        
-        if (player.level() != chunk.getLevel()) {
-            setSyncedState(player, pos, false);
-            return;
-        }
-
         int minY = chunk.getMinSectionY();
         List<LODDataPayload.SectionData> sections = new ArrayList<>();
         

